tests/test_embedding.py

```diff
@@ -0,0 +1,79 @@
+from __future__ import annotations
+
+import numpy as np
+import torch
+
+from qadence_embeddings.callable import ConcretizedCallable
+from qadence_embeddings.embedding import Embedding
+
+
+def test_embedding() -> None:
+    x = np.random.uniform(0, 1)
+    theta = np.random.uniform(0, 1)
+    results = []
+    for engine_name in ["jax", "torch", "numpy"]:
+        v_params = ["theta"]
+        f_params = ["x"]
+        leaf0, native_call0 = "%0", ConcretizedCallable(
+            "mul", ["x", "theta"], {}, engine_name
+        )
+        embedding = Embedding(
+            v_params,
+            f_params,
+            None,
+            var_to_call={leaf0: native_call0},
+            engine_name=engine_name,
+        )
+        inputs = {
+            "x": (torch.tensor(x) if engine_name == "torch" else x),
+            "theta": (torch.tensor(theta) if engine_name == "torch" else theta),
+        }
+        eval_0 = embedding.var_to_call["%0"](inputs)
+        results.append(eval_0.item())
+    assert np.allclose(results[0], results[1]) and np.allclose(results[0], results[2])
+
+
+def test_reembedding() -> None:
+    x = np.random.uniform(0, 1)
+    theta = np.random.uniform(0, 1)
+    t = np.random.uniform(0, 1)
+    t_reembed = np.random.uniform(0, 1)
+    results = []
+    reembedded_results = []
+    for engine_name in ["jax", "torch", "numpy"]:
+        v_params = ["theta"]
+        f_params = ["x"]
+        tparam = "t"
+        leaf0, native_call0 = "%0", ConcretizedCallable(
```

Comment:
Would it make sense to randomly draw from a poll of functions and variable names to construct these instructions ? I am a bit worried that we only cover a very reduced subset of the possibilities.